homeassistant/components/stookwijzer/sensor.py

```diff
@@ -1,20 +1,77 @@
-"""Support for Stookwijzer Sensor."""
-
+"""Support for Stookwijzer Sensors."""
 from __future__ import annotations
 
-from datetime import timedelta
-
-from stookwijzer import Stookwijzer
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import cast
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfSpeed
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN, StookwijzerState
+from .coordinator import StookwijzerCoordinator, StookwijzerData
+from .entity import StookwijzerEntity
+
+
+@dataclass(frozen=True)
+class StookwijzerSensorDescriptionMixin:
+    """Required values for Stookwijzer sensors."""
+
+    value_fn: Callable[[StookwijzerCoordinator], int | float | str | None]
+    attr_fn: Callable[[StookwijzerCoordinator], list | None] | None
+
+
+@dataclass(frozen=True)
+class StookwijzerSensorDescription(
+    SensorEntityDescription,
+    StookwijzerSensorDescriptionMixin,
+):
+    """Class describing Stookwijzer sensor entities."""
```

Comment:
Same as the binary sensor example, this can be simplified to a single class now.